pygmt/clib/conversion.py

```diff
@@ -313,6 +320,11 @@ def array_to_datetime(array: Sequence[Any]) -> np.ndarray:
 
     If the input array is not in legal datetime formats, raise a ValueError exception.
 
+    .. deprecated:: 0.14.0
```

Comment:
After this PR, `array_to_datetime` is no longer used, but I still want to keep this function so that we know what kinds of datetime formats that `np.asarray(array, dtype=np.datetime64)` can support.